test/fail_compilation/retscope.d

```diff
@@ -349,3 +349,37 @@ int* bar11(scope int** x) @safe
 }
 
 int* foo11(int* x) @safe { return x; }
+
+/******************************************/
+
+/*
+TEST_OUTPUT:
+---
+fail_compilation/retscope.d(366): Error: escaping reference to local variable f
+fail_compilation/retscope.d(377): Error: scope variable f may not be returned
+---
+*/
+
+int* escape12() @safe
+{
+    Foo12 f;
+    return f.foo;
+}
+
+struct Foo12
+{
+    int* foo() return @safe;
+}
+
+int* escape13() @safe
+{
+    scope Foo13 f;
```

Comment:
I would hold off on any `scope class` test before the fix in stable gets merged into scope. @MartinNowak ?